pkg/lockservice/types.go

```diff
@@ -56,11 +56,15 @@ const (
 type LockStorage interface {
 	// Add we use kv to store the lock. Key is a locked row or a row range. Value is the
 	// TxnID.
-	Add(key, value []byte)
+	Add(key []byte, value Lock)
```

Comment:
dont you need to expose lock mode here?

Reply:
lock mode is in the lock struct.  The value field in the lock struct contains lock mode and lock granularity

Reply:
how so -- how Lock has another value, as a byte.    How caller can setup the right value?   Your plan to write a Lock constructor or something later?